### PR TITLE
refactor(chip): extract sub components into styled-jsx style components

### PR DESCRIPTION
**Pull request by Mohammer5**
I'm creating this PR as a draft PR because I'm not sure this is the right direction for these components.
It's just a gut feeling, but it feels little weird, splitting the components like this just to apply certain styles, I lack a better alternative that matches our criteria right now though..
Anyways, this PR decreases the complexity of the `Chip` component itself.

In the second commit (2e72e83) I removed some style rules that haven't been used, there's no `.icon` class anywhere. I guess this is a leftover from when @varl switched to styled-jsx, as the icon components (passed to `Chip` via the `icon` prop) could have the `.icon` class name. But with the styled-jsx approach this style wouldn't be applied anyways. How should we handle that style? In my opinion the style should be applied to the icon component by the app itself rather than by this library, so there wouldn't be a need to change the current implementation of `Chip`

### CODE REVIEW

File: src/Chip/Remove.js [hunk 1]
```diff
@@ -29,7 +30,7 @@ export class Remove extends PureComponent {
     render() {
         const onRemove = this.props.onRemove
 
-        if (!onRemove) return
+        if (!onRemove) return null
```
Comment (varl) [inline src/Chip/Remove.js]: Why `null` instead of `undefined`?

Comment (Mohammer5) [reply src/Chip/Remove.js]: I think it doesn't really make a difference, but interestingly I wanted to know if there's any official docs about this (I usually returned undefined so far) and found this:
https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

So I just copied that

Comment (varl) [reply src/Chip/Remove.js]: Oh, that's interesting, it _does_ affect the CodePen linked: https://codepen.io/gaearon/pen/Xjoqwm?editors=0010

Try switching the `return null` to `return`. Then hit the `Hide` button. Now it's gone completely whereas if you return `null` the button is still there. React treats them differently, one stops rendering and one doesn't.

>Returning null from a component’s render method does not affect the firing of the component’s lifecycle methods. For instance `componentDidUpdate` will still be called.

Comment (Mohammer5) [reply src/Chip/Remove.js]: > Now it's gone completely whereas if you return null the button is still there.

Huh... That's VERY weird behavior, will never use undefined again!

**Pull request merged**